Sources/Fog/FogSyncChecker.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+import Foundation
+
+protocol FogSyncCheckable {
+    var viewsHighestKnownBlock: UInt64 { get }
+    var ledgersHighestKnownBlock: UInt64 { get }
+    var consensusHighestKnownBlock: UInt64 { get }
+    var currentBlockIndex: UInt64 { get }
+    var fogSyncThreshold: UInt64 { get }
+    
+    func inSync() -> Result<(), FogSyncError>
+    func setLedgersHighestKnownBlock(_:UInt64)
+    func setViewsHighestKnownBlock(_:UInt64)
+    func setConsensusHighestKnownBlock(_:UInt64)
+}
+
+extension FogSyncCheckable {
+    func inSync() -> Result<(), FogSyncError> {
+        guard viewLedgerOutOfSync == false else {
+            return .failure(.viewLedgerOutOfSync(viewsHighestKnownBlock, ledgersHighestKnownBlock))
+        }
+        guard consensusOutOfSync == false else {
+            return .failure(.consensusOutOfSync(consensusHighestKnownBlock, currentBlockIndex))
+        }
+        return .success(())
+    }
+    
+    var currentBlockIndex: UInt64 {
+        min(ledgersHighestKnownBlock, viewsHighestKnownBlock)
+    }
+    
+    private var viewLedgerOutOfSync: Bool {
+        abs(
+            Int64(
+                max(viewsHighestKnownBlock, ledgersHighestKnownBlock)
```

Comment:
using possessive 's' is a bit strange in variable names b/c no apostrophe, would prefer highestKnownViewBlock highestKnownLedgerBlock

Reply:
I like how the first work unambiguates tho

Reply:
sorry - what do you mean first work?

Reply:
corrected:

> I like how the first word disambiguates tho

```
viewXXX
ledgerXXX
consensusXXX
```

Reply:
maybe just remove the "s" to make them "viewHighestBlock" and "ledgerHighestKnownBlock"?  The only thing bad about that is that "view" is both a noun and a verb, but if it's consistent with the rest of the code, that should be at least as ok as the code already is.

Reply:
I like the readability of the possesive, lets revisit this type of situation when we work on the linting rules, thats one of the goals for this quarter and we might be able to enforce possesive/non-possesive w/ a grep rule in swiftlint.